unit/java_bytecode/java_bytecode_specialize_generics/generics.java

```diff
@@ -0,0 +1,61 @@
+public class generics {
+
+  class element<E> {
+    E elem;
+  }
+
+  class bound_element<NUM extends java.lang.Number> {
+    NUM elem;
+    NUM f() {
+      return elem;
+    }
+
+    void g(NUM e) {
+      elem=e;
+    }
+
+  }
+
+  bound_element<Integer> belem;
+
+  Integer f(int n) {
+
+    element<Integer> e=new element<>();
+    e.elem=n;
+    bound_element<Integer> be=new bound_element<>();
+    belem=new bound_element<>();
+    be.elem=new Integer(n+1);
+    ;
+    if(n>0)
+      return e.elem;
+    else
+      return be.elem;
+  }
+
+  class double_element<A,B> {
+  	A first;
+  	B second;
+    java.util.Map<A,B> map;
```

Comment:
I am concerned using the library classes like `Map` and `List` as I suppose it is just stubbing them which may not be exactly what we want.

Reply:
Ah, good point - I'll update the tests to avoid standard container classes.